kratos/python/add_strategies_to_python.cpp

```diff
@@ -299,55 +301,72 @@ namespace Kratos
-	    typedef ResidualBasedNewmarkDisplacementScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedNewmarkDisplacementSchemeType;
-
-            class_< BaseSchemeType, boost::noncopyable >
-                    ("Scheme", init< >())
-                    .def("Initialize", &BaseSchemeType::Initialize)
-                    .def("SchemeIsInitialized", &BaseSchemeType::SchemeIsInitialized)
-                    .def("ElementsAreInitialized", &BaseSchemeType::ElementsAreInitialized)
-                    .def("ConditionsAreInitialized", &BaseSchemeType::ConditionsAreInitialized)
-                    .def("InitializeElements", &BaseSchemeType::InitializeElements)
-                    .def("InitializeConditions", &BaseSchemeType::InitializeConditions)
-                    .def("InitializeSolutionStep", &BaseSchemeType::InitializeSolutionStep)
-                    .def("FinalizeSolutionStep", &BaseSchemeType::FinalizeSolutionStep)
-                    .def("InitializeNonLinIteration", &BaseSchemeType::InitializeNonLinIteration)
-                    .def("FinalizeNonLinIteration", &BaseSchemeType::FinalizeNonLinIteration)
-                    .def("Predict", &BaseSchemeType::Predict)
-                    .def("Update", &BaseSchemeType::Update)
-                    .def("CalculateOutputData", &BaseSchemeType::CalculateOutputData)
-                    .def("Clean", &BaseSchemeType::Clean)
-                    .def("Clear",&BaseSchemeType::Clear)
-                    .def("MoveMesh", MoveMesh)
-                    .def("Check", &BaseSchemeType::Check)
-                    ;
-
-            class_< ResidualBasedIncrementalUpdateStaticScheme< SparseSpaceType, LocalSpaceType>,
-                    bases< BaseSchemeType >, boost::noncopyable >
-                    (
-                    "ResidualBasedIncrementalUpdateStaticScheme", init< >()
-                    );
-
-            class_< ResidualBasedIncrementalUpdateStaticSchemeSlip< SparseSpaceType, LocalSpaceType>,
-                    bases< ResidualBasedIncrementalUpdateStaticScheme< SparseSpaceType, LocalSpaceType> >,
-                    boost::noncopyable >
-                    ("ResidualBasedIncrementalUpdateStaticSchemeSlip", init<unsigned int, unsigned int>());
-
-	    // Residual Based Bossak Scheme Type
-	    class_< ResidualBasedBossakDisplacementSchemeType,
+        typedef ResidualBasedBossakDisplacementScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedBossakDisplacementSchemeType;
+        typedef ResidualBasedNewmarkDisplacementScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedNewmarkDisplacementSchemeType;
+        typedef ResidualBasedBDFDisplacementScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedBDFDisplacementSchemeType;
+        typedef ResidualBasedBDFCustomScheme< SparseSpaceType, LocalSpaceType >  ResidualBasedBDFCustomSchemeType;
+
+        class_< BaseSchemeType, boost::noncopyable >
```

Comment:
you don't need (better, you should not) redefine all the methods. just add the basis class and have boost.python to inherit them for you

(i appreciate that you just copied what was there... consider it as an improvement)

Reply:
I will clean it

Reply:
Huuum, this is the base scheme type, should have all that definitions